### PR TITLE
Magic Window Demo

### LINKED ISSUE #613: Add magnifying glass example
Came across this and thought it'd be cool to try to pull off with our maps

screen shot 2018-01-23 at 10 34 17 am

### PR DESCRIPTION
**Pull request by sansumbrella**
Adds magic window demo showing two maps kept in sync and alpha masking with the Android view system.

screenshot_1539370397
screenshot_1539370403
screenshot_1539370425

Closes #613

### CODE REVIEW

**Comment by sansumbrella** [comment]
### Questions for reviewers

What kinds of tests do we typically add for examples?
Class organization? All classes are currently in one file to keep the demo together.
What linter do you run locally (to avoid back-and-forth with circleci)?

**Comment by langsmith** [comment]
ezgif com-resize

**Comment by langsmith** [comment]
Hell of an example @sansumbrella . 👏 🎊  Made a commit that cleans up some small stuff; mainly moving string resource files to their correct locations.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MagicWindowActivity.kt [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import kotlinx.android.synthetic.main.activity_magic_window.*
+import kotlin.math.max
+import kotlin.math.min
+
+const val TAG = "DragActivityTag"
+const val PERMISSION_REQUEST_LOCATION = 404
+const val BASE_MAP_BUNDLE = "$TAG.basemap.bundle"
+const val REVEAL_MAP_BUNDLE = "$TAG.revealedMap.bundle"
+
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MagicWindowActivity.kt]: Please add javadocs to explain what this example is about. E.g:

```
/**
 * This example...
 */
```

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MagicWindowActivity.kt [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import kotlinx.android.synthetic.main.activity_magic_window.*
+import kotlin.math.max
+import kotlin.math.min
+
+const val TAG = "DragActivityTag"
+const val PERMISSION_REQUEST_LOCATION = 404
+const val BASE_MAP_BUNDLE = "$TAG.basemap.bundle"
+const val REVEAL_MAP_BUNDLE = "$TAG.revealedMap.bundle"
```
Comment (langsmith) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/MagicWindowActivity.kt]: For these `const val`, please move them within the activity below the `var initialZoom` line with 

```kotlin
companion object {
        const val TAG = "DragActivityTag"
        const val PERMISSION_REQUEST_LOCATION = 404
        const val BASE_MAP_BUNDLE = "$TAG.basemap.bundle"
        const val REVEAL_MAP_BUNDLE = "$TAG.revealedMap.bundle"
}
```

**Comment by langsmith** [comment]
>What kinds of tests do we typically add for examples?

Not much. We run a test that makes sure that the example can be opened and doesn't crash:
  - https://github.com/mapbox/mapbox-android-demo/blob/master/Makefile#L7
  - https://github.com/mapbox/mapbox-android-demo/blob/master/scripts/generate-test-code.js
    - Haven't thought about how/whether the `.js` file above needs to be changed now that Kotlin examples are being added

[We also do Firebase instrumentation tests](https://github.com/mapbox/mapbox-android-demo/blob/master/circle.yml#L56-L66)

>Class organization? All classes are currently in one file to keep the demo together.

I'm not sure what the question/concern here is. A little clarification?

>naming: I noticed the other kotlin demo currently has Kotlin in the file/classname

Yes, let's have Kotlin examples have the word `Kotlin` in their names.

>What linter do you run locally (to avoid back-and-forth with circleci)?

We don't? Don't think we do. Running `make checkstyle` in Terminal will check for checkstyle issues. Was there a specific reason why you were asking about this?

**Comment by sansumbrella** [comment]
Should be ready for another round of review and merging! Thanks.

**Comment by langsmith** [comment]
> keep all code for one demo in a single file

For now, let's do this. So keep things as is. 


`option` + `command` + `l` in Android Studio does autoformatting, which helps for checkstyle

**Pull request merged**